src/react-integration/NetworkErrorBoundary.tsx

```diff
@@ -1,27 +1,32 @@
 import React from 'react';
 
+export interface ErrorWithStatus extends Error {
+  status: number;
+  response?: { statusText?: string, body?: any };
+}
+
 interface Props {
   children: React.ReactNode;
-  fallbackComponent: React.ComponentType<{ error: Error }>;
+  fallbackComponent: React.ComponentType<{ error: ErrorWithStatus }>;
 }
 interface State {
-  error?: Error;
+  error?: ErrorWithStatus;
 }
 export default class NetworkErrorBoundary extends React.Component<
```

Comment:
To make this even more reusable you might even use a generic component (https://www.mojotech.com/blog/typescript-generic-react-components/)

```typescript
interface Props<E> {
  children: React.ReactNode;
  fallbackComponent: React.ComponentType<{ error: E }>;
}
interface State<E> {
  error?: E;
}
export default class NetworkErrorBoundary<E extends ErrorWithStatus = ErrorWithStatus>
```